packages/fx-core/src/plugins/resource/apiconnector/envHandler.ts

```diff
@@ -30,24 +30,42 @@ export class ApiDataManager {
     if (!this.apiConnector[apiName]) {
       this.apiConnector[apiName] = new Map();
     }
+    const endPoint = Constants.envPrefix + apiName + "_ENDPOINT";
+    const authName = Constants.envPrefix + apiName + "_AUTHENTICATION_TYPE";
+    this.apiConnector[apiName].set(authName, config.AuthConfig.AuthType);
+    this.apiConnector[apiName].set(endPoint, config.EndPoint);
     if (config.AuthConfig.AuthType === AuthType.BASIC) {
       this.addBasicEnvs(config);
+    } else if (config.AuthConfig.AuthType === AuthType.AAD) {
+      this.addAADEnvs(config);
     }
   }
 
   public addBasicEnvs(config: ApiConnectorConfiguration) {
     const apiName: string = config.APIName.toUpperCase();
     const apiConfig = this.apiConnector[apiName];
-    const endPoint = "API_" + apiName + "_ENDPOINT";
-    const authName = "API_" + apiName + "_AUTHENTICATION_TYPE";
-    const userName = "API_" + apiName + "_USERNAME";
-    const passWd = "API_" + apiName + "_PASSWORD";
+    const userName = Constants.envPrefix + apiName + "_USERNAME";
+    const passWd = Constants.envPrefix + apiName + "_PASSWORD";
     const authConfig = config.AuthConfig as BasicAuthConfig;
     apiConfig.set(userName, authConfig.UserName);
-    apiConfig.set(authName, authConfig.AuthType);
-    apiConfig.set(endPoint, config.EndPoint);
     apiConfig.set(passWd, "");
   }
+
+  public addAADEnvs(config: ApiConnectorConfiguration) {
+    const apiName: string = config.APIName.toUpperCase();
+    const apiConfig = this.apiConnector[apiName];
+    const reuseTeamsApp = Constants.envPrefix + apiName + "_REUSE_TEAMS_APP";
```

Comment:
will this be added to `.env` file? we may need to exclude this config since it's internal only.

Reply:
@wenytang-ms will use this config to decide the notification message.
If we do not write it in env, we may use the existing app's config?